File: zig/resistor-color/resistor_color.zig
pub const ColorBand = enum(u4) {
    black,
    brown,
    red,
    orange,
    yellow,
    green,
    blue,
    violet,
    grey,
    white,
};

pub fn colorCode(color: ColorBand) usize {
    return @intFromEnum(color);
}

pub fn colors() []const ColorBand {
    return &[10]ColorBand{
        .black,
        .brown,
        .red,
        .orange,
        .yellow,
        .green,
        .blue,
        .violet,
        .grey,
        .white,
    };
}

File: go/resistor-color/resistor_color.go
package resistorcolor

// Colors returns the list of all colors.
var colors = map[string]int{
	"black":  0,
	"brown":  1,
	"red":    2,
	"orange": 3,
	"yellow": 4,
	"green":  5,
	"blue":   6,
	"violet": 7,
	"grey":   8,
	"white":  9,
}

func Colors() []string {
	c := make([]string, len(colors))

	for k, v := range colors {
		c[v] = k
	}

	return c
}

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	return colors[color]
}
